rln/src/protocol.rs

```diff
@@ -396,6 +414,21 @@ pub fn keygen() -> (Fr, Fr) {
     (identity_secret, id_commitment)
 }
 
+// Generates a tuple (id_trapdoor, id_nullifier, id_secret, id_commitment) where
+// id_trapdoor and id_nullifier are random,
+// identity_secret = PoseidonHash(id_trapdoor, id_nullifier),
+// id_commitment = PoseidonHash(identity_secret),
+// RNG is instantiated using thread_rng()
+// Generated credentials are compatible with Semaphore credentials
+pub fn extended_keygen() -> (Fr, Fr, Fr, Fr) {
+    let mut rng = thread_rng();
+    let id_trapdoor = Fr::rand(&mut rng);
+    let id_nullifier = Fr::rand(&mut rng);
+    let identity_secret = poseidon_hash(&[id_trapdoor, id_nullifier]);
```

Comment:
I think this should be renamed to `id_secret_hash` / `identity_secret_hash` to make it clearer according to https://rfc.vac.dev/spec/32/#identity_secret

Reply:
Thanks!